src/azul/plugins/metadata/anvil/__init__.py

```diff
@@ -277,9 +280,20 @@ def manifest_config(self) -> ManifestConfig:
         # the field mapping. Keys are field paths in an ES hit, and values are
         # the desired manifest column name, or None to omit the column from the
         # manifest.
+        #
+        # Note that there is a brittle coupling that must be maintained between
+        # the keys here and those used in `self._field_mapping`. Also, the
+        # values (that aren't `None`) should match the related field's path in
+        # a response hit from the `/index/files` endpoint.
+        #
         custom_field_names = {
+            ('bundles', 'uuid'): 'bundles.bundle_uuid',
```

Comment:
This continues on the vein of making the code self-documenting (from the previous patch provided) and having clear separation of responsibilities. You may still want to document the brittleness, or isolated in the code for the documentation to be inlined in the code, which makes it harder to miss, in my opinion.

```diff
diff --git a/src/azul/plugins/metadata/anvil/__init__.py b/src/azul/plugins/metadata/anvil/__init__.py
index 18d5a657..ad9060fa 100644
--- a/src/azul/plugins/metadata/anvil/__init__.py
+++ b/src/azul/plugins/metadata/anvil/__init__.py
@@ -276,26 +276,19 @@ class Plugin(MetadataPlugin[AnvilBundle]):
     def manifest_config(self) -> ManifestConfig:
         result = defaultdict(dict)
 
-        # This is used to modify the field names from how they are specified in
-        # the field mapping. Keys are field paths in an ES hit, and values are
-        # the desired manifest column name, or None to omit the column from the
-        # manifest.
-        #
-        # Note that there is a brittle coupling that must be maintained between
-        # the keys here and those used in `self._field_mapping`. Also, the
-        # values (that aren't `None`) should match the related field's path in
-        # a response hit from the `/index/files` endpoint.
-        #
-        custom_field_names = {
-            ('bundles', 'uuid'): 'bundles.bundle_uuid',
-            ('bundles', 'version'): 'bundles.bundle_version',
+        fields_to_omit_in_manifest = {
             ('contents', 'activities', 'activity_table'): None,
             ('contents', 'files', 'uuid'): None,
             ('contents', 'files', 'version'): None,
+        }
+        field_path_to_manifest_column = {
+            # These values should match the related field's path in a response
+            # hit from the `/index/files` endpoint.
+            ('bundles', 'uuid'): 'bundles.bundle_uuid',
+            ('bundles', 'version'): 'bundles.bundle_version',
             ('sources', 'id'): 'sources.source_id',
             ('sources', 'spec'): 'sources.source_spec',
         }
-
         def recurse(mapping: MetadataPlugin._FieldMapping, path: FieldPath):
             for path_element, name_or_type in mapping.items():
                 new_path = (*path, path_element)
@@ -304,15 +297,18 @@ class Plugin(MetadataPlugin[AnvilBundle]):
                 elif isinstance(name_or_type, str):
                     if new_path == ('entity_id',):
                         pass
-                    elif new_path in custom_field_names:
-                        result[path][path_element] = custom_field_names.pop(new_path)
+                    elif new_path in fields_to_omit_in_manifest:
+                        result[path][path_element] = fields_to_omit_in_manifest.pop(new_path)
+                    elif new_path in field_path_to_manifest_column:
+                        result[path][path_element] = field_path_to_manifest_column.pop(new_path)
                     else:
                         result[path][path_element] = name_or_type
                 else:
                     assert False, (path, path_element, name_or_type)
 
         recurse(self._field_mapping, ())
-        assert len(custom_field_names) == 0, custom_field_names
+        assert len(fields_to_omit_in_manifest) == 0, fields_to_omit_in_manifest
+        assert len(field_path_to_manifest_column) == 0, field_path_to_manifest_column
         # The file URL is synthesized from the `uuid` and `version` fields.
         # Above, we already configured these two fields to be omitted from the
         # manifest since they are not informative to the user.

```

Consider making the assertion failure a tad more informative.